indico/modules/rb_new/util.py

```diff
@@ -541,3 +542,33 @@ def serialize_nonbookable_periods(data):
 
 def serialize_unbookable_hours(data):
     return [bookable_hours_schema.dump(d).data for d in data]
+
+
+def get_room_blockings(user):
+    return Blocking.query.filter(Blocking.created_by_user == user).all()
+
+
+def create_blocking(rooms, start_date, end_date, reason, allowed_principals):
+    blocking = Blocking()
+    blocking.start_date = start_date
+    blocking.end_date = end_date
+    blocking.reason = reason
+    blocking.created_by_user = session.user
+    blocking.allowed = allowed_principals
+    blocking.blocked_rooms = [BlockedRoom(room_id=room.id) for room in rooms]
+    db.session.add(blocking)
+    db.session.flush()
+    return blocking
+
+
+def approve_blocking(blocking):
```

Comment:
I understand the fact that this method will "approve" a blocking if the user is allowed. However, it's not always the case. What about `register_blocking(...)` or similar? or even `approve_or_request_blocking(...)`...